### PR TITLE
RBAC: drop permissions with folder:create action added by migration

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
In the dashboard and folders migration folder:create action was added by mistake.
In this pr a new migration is added to drop all folders:create permissions added by previous migration

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go [hunk 1]
```diff
@@ -280,3 +279,18 @@ func (d *dashboardUidPermissionMigrator) migrateIdScopes(sess *xorm.Session) err
...
+type managedFolderCreateAction struct {
+	migrator.MigrationBase
+}
+
+func (m *managedFolderCreateAction) SQL(dialect migrator.Dialect) string {
+	return CodeMigrationSQL
+}
+
+func (m *managedFolderCreateAction) Exec(sess *xorm.Session, migrator *migrator.Migrator) error {
+	if _, err := sess.Exec("DELETE FROM permission WHERE action = 'folders:create' AND scope LIKE 'folders:uid:%'"); err != nil {
```
Comment (IevaVasiljeva) [inline pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:292]: question: I guess we're assuming that for custom roles with `folders:create` action the scope won't be applied? Just want to make sure that we're not potentially messing up custom roles with this migration.

File: pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go [hunk 2]
```diff
@@ -41,13 +41,11 @@ var folderPermissionTranslation = map[models.PermissionType][]string{
 		dashboards.ActionDashboardsCreate,
 		dashboards.ActionFoldersRead,
 		dashboards.ActionFoldersWrite,
-		dashboards.ActionFoldersCreate,
```
Comment (IevaVasiljeva) [inline pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:44]: nit: I don't think it really matters in this case, but we shouldn't change existing migrations.

File: pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go [hunk 3]
```diff
@@ -57,6 +55,7 @@ var folderPermissionTranslation = map[models.PermissionType][]string{
 func AddDashboardPermissionsMigrator(mg *migrator.Migrator) {
 	mg.AddMigration("dashboard permissions", &dashboardPermissionsMigrator{})
 	mg.AddMigration("dashboard permissions uid scopes", &dashboardUidPermissionMigrator{})
+	mg.AddMigration("drop managed folder create actions", &managedFolderCreateAction{})
```
Comment (IevaVasiljeva) [inline pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:58]: nit: again, I don't think it matters in this case, but we should aim to add new migrations to the end of the full migration list to ensure the migration order.

**Review by IevaVasiljeva — COMMENTED** [review]
LGTM migration wise, but I'd probably not change the existing migration and add the new migration to the end of the list in https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/migrations/migrations.go

File: pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go (hunk 3 above)
Comment (kalleep) [reply pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:58]: It does not matter in this case, we just need to make sure we runt this after the other two

**Review by IevaVasiljeva — APPROVED** [review]

**Comment by gamab** [comment]
Should we backport this to 8.5.x?

**Pull request merged**